HIP/hip-21.md

```diff
@@ -32,41 +33,70 @@ Today, applications have to use the File Service and submit two file contents qu
 
 ## Rationale 
 
-The proposal is to create a free non-file HAPI query to return network information. As we continue to grow the network and transition into a permissionless model, the number of times the local address book will be required to update will increase. The transaction fee associated with the current file query hinders the ability to update the address book as frequently as it might be needed for applications to ensure their address book is always up to date. This information should be free. 
+The proposal is to create a rest API query to return network information. As we continue to grow the network and transition into a permissionless model, the number of times the local address book will be required to update will increase. The transaction fee associated with the current file query hinders the ability to update the address book as frequently as it might be needed for applications to ensure their address book is always up to date. This information should be free. 
 
 ## Specification 
 
-- Add a non-file HAPI query called **getNetworkInfo** to the existing Network Service
-- This query would not have a transaction fee
+REST API
+
+GET `api/v1/addressbooks`
+
+Response:
+
+```json
+
+{
+  "addressbook": [
+    {
+      "ipAddressv2": {
+      },
+      "port": " " ,
+      "rsa_publickey": "  ",
+      "node_id": " ",
+      "node_account_id": " ",
+      "node_cert_hash": ,
+      "memo": " ",
+      "description": " ",
+      "stake": " ",
+    }
+  ],
+  "links": {
+    "next": null
+  }
+}
+
+```
+
+Example with all params:
+api/v1/addressbooks/timestamp=gte:1566562500.040961001&timestamp=lt:1576562500.040961001&file.id=0.0.102&limit=5&order=asc
+
 
 
-[Network Service](https://github.com/hashgraph/hedera-protobufs/blob/main/services/NetworkService.proto)
+## Security Implications
+
+- A client could spam the network since the query is free and disable other users from processing their request 
```

Comment:
Could add that this could be mitigated by rate limiting, limiting connections per IP and caching the address book response.